contracts/src/base/registry/facets/distribution/v2/RewardsDistribution.sol

```diff
@@ -0,0 +1,421 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+// interfaces
+import {IERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Permit.sol";
+import {IRewardsDistribution} from "./IRewardsDistribution.sol";
+
+// libraries
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {SignatureCheckerLib} from "solady/utils/SignatureCheckerLib.sol";
+import {CustomRevert} from "contracts/src/utils/libraries/CustomRevert.sol";
+import {NodeOperatorStorage} from "contracts/src/base/registry/facets/operator/NodeOperatorStorage.sol";
+import {SpaceDelegationStorage} from "contracts/src/base/registry/facets/delegation/SpaceDelegationStorage.sol";
+import {StakingRewards} from "./StakingRewards.sol";
+import {RewardsDistributionStorage} from "./RewardsDistributionStorage.sol";
+
+// contracts
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+import {OwnableBase} from "contracts/src/diamond/facets/ownable/OwnableBase.sol";
+import {Nonces} from "contracts/src/diamond/utils/Nonces.sol";
+import {EIP712Base} from "contracts/src/diamond/utils/cryptography/signature/EIP712Base.sol";
+
+contract RewardsDistribution is
+  IRewardsDistribution,
+  OwnableBase,
+  EIP712Base,
+  Nonces,
+  Facet
+{
+  using EnumerableSet for EnumerableSet.AddressSet;
+  using StakingRewards for StakingRewards.Layout;
+
+  bytes32 internal constant STAKE_TYPEHASH =
+    keccak256(
+      "Stake(uint96 amount,address delegatee,address beneficiary,address owner,uint256 nonce,uint256 deadline)"
+    );
+
+  /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
```

Comment:
why are we using this comment style?

Reply:
Because it's [customary](https://github.com/Vectorized/solady/blob/main/src/tokens/ERC20.sol#L22-L24).